chiselFrontend/src/main/scala/chisel3/core/BlackBox.scala

```diff
@@ -17,6 +17,84 @@ case class StringParam(value: String) extends Param
 /** Unquoted String */
 case class RawParam(value: String) extends Param
 
+abstract class BaseBlackBox extends BaseModule
+
+/** Defines a black box, which is a module that can be referenced from within
+  * Chisel, but is not defined in the emitted Verilog. Useful for connecting
+  * to RTL modules defined outside Chisel.
+  *
+  * A variant of BlackBox, this has a more consistent naming scheme in allowing
+  * multiple top-level IO and does not drop the top prefix.
+  *
+  * @example
+  * Some design require a differential input clock to clock the all design.
+  * With the xilinx FPGA for example, a Verilog template named IBUFDS must be
+  * integrated to use differential input:
+  * {{{
+  *  IBUFDS #(.DIFF_TERM("TRUE"),
+  *           .IOSTANDARD("DEFAULT")) ibufds (
+  *   .IB(ibufds_IB),
+  *   .I(ibufds_I),
+  *   .O(ibufds_O)
+  *  );
+  * }}}
+  *
+  * To instantiate it, a BlackBox can be used like following:
+  * {{{
+  * import chisel3._
+  * import chisel3.experimental._
+  *
+  * // Example with Xilinx differential buffer IBUFDS
+  * class IBUFDS extends ExtModule(Map("DIFF_TERM" -> "TRUE", // Verilog parameters
+  *                                    "IOSTANDARD" -> "DEFAULT"
+  *                      )) {
+  *   val O = IO(Output(Clock()))
+  *   val I = IO(Input(Clock()))
+  *   val IB = IO(Input(Clock()))
+  * }
+  * }}}
+  * @note The parameters API is experimental and may change
+  */
+abstract class ExtModule(val params: Map[String, Param] = Map.empty[String, Param]) extends BaseBlackBox {
+  private[core] override def generateComponent(): Component = {
+    require(!_closed, "Can't generate module more than once")
+    _closed = true
+
+    val names = nameIds(classOf[ExtModule])
+
+    // Name ports based on reflection
```

Comment:
I refuse to accept the claim that use of reflection has not been a problem. One came up with Bundles (which classically have less implementation trickery than modules since users would come to understand that reflection was being used to inspect them) literally days before you proposed expanding the use of reflection (#456).